app/locales/languages.ts

```diff
@@ -130,26 +136,39 @@ export const initDevLanguages = () => {
 };
 
 /** The known locale list */
-export const getLocaleList = () =>
-  Object.entries(i18next.options.resources)
+export const getLocaleList = (): Array<{
+  value: Locale;
+  label: string;
+}> => {
+  const resources = i18next.options.resources;
+  if (!resources) {
+    return [];
+  }
+  return Object.entries(resources)
     .map(([langCode, lang]) => ({
       value: langCode,
-      label: lang.label,
+      label: lang.label as string,
```

Comment:
any way we can avoid this type casting here?  I would tackle this in a separate pr maybe, but seems like `Object.entries` like a code smell?  Why do we even need to do this conversion? What's the difference between `value` and `label` here?

Reply:
The return value of the resources method returns a union type called `ResourceKey` that looks like this:

``` typescript

type ResourceKey =
  | string
  | {
      [key: string]: any;
    };
```

We need to decide which type of the union to use and the API exposed here is consumed with the assumption all of those values are strings, so someone needs to do the casting whether here or on the consumers. Happy to shift it but someone will cast it 😓